src/components/MasterItemsSelect.js

```diff
@@ -51,9 +56,9 @@ class MasterItemsSelect extends Component {
           { className: 'form-group' },
           dom.label(null, 'Item'),
           dom.select(
-            { className: 'select-items form-control input-lg', onChange: this.handleChange },
+            { className: 'select-item form-control input-lg', onChange: this.handleChange, value: this.state.selectedItemId },
```

Comment:
Why do we have to explicitly set the value now? Isn't it just the value of the initially selected option?